src/pages/settings/subscriptions/create/Create.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { AxiosError } from 'axios';
+import { endpoint } from 'common/helpers';
+import { request } from 'common/helpers/request';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { GenericSingleResourceResponse } from 'common/interfaces/generic-api-response';
+import { Product } from 'common/interfaces/product';
+import { Subscription } from 'common/interfaces/subscription';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { useProductsQuery } from 'common/queries/products';
+import { Settings } from 'components/layouts/Settings';
+import { TabGroup } from 'components/TabGroup';
+import { FormEvent, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from 'react-query';
+import { useNavigate } from 'react-router-dom';
+import { Overview } from '../common/components/Overview';
+import { Settings as SubscriptionSettings } from '../common/components/Settings';
+import { Webhook } from '../common/components/Webhook';
+import { useBlankSubscriptionQuery } from '../common/hooks/useBlankSubscriptionQuery';
+import { useHandleChange } from '../common/hooks/useHandleChange';
+
+export function Create() {
+  const [t] = useTranslation();
+
+  const navigate = useNavigate();
+
+  const { data } = useBlankSubscriptionQuery();
+
+  const { data: productsData } = useProductsQuery();
+
+  const queryClient = useQueryClient();
+
+  const pages = [
+    { name: t('settings'), href: '/settings' },
+    { name: t('subscriptions'), href: '/settings/subscriptions' },
+    { name: t('new_subscription'), href: '/settings/subscriptions/create' },
+  ];
+
+  const tabs = [t('overview'), t('settings'), t('webhook')];
+
+  const [subscription, setSubscription] = useState<Subscription>();
+
+  const [products, setProducts] = useState<Product[]>();
+
+  const [errors, setErrors] = useState<ValidationBag>();
+
+  const handleChange = useHandleChange({
+    setErrors,
+    setSubscription,
+    subscription,
+  });
+
+  useEffect(() => {
+    if (data) {
+      setSubscription({
+        ...data,
+        frequency_id: '5',
```

Comment:
Can we please update this with the enums and not use plain numbers/strings?

Reply:
Sure, adjusted